api/scripts/certification/fix-validated-live-alert-certification-challenge-ids.js

```diff
@@ -0,0 +1,198 @@
+
+      // Now fix capacities
+      const capacitiesToUpdate = [];
+      for (const currentCapacity of certificationCapacities) {
+        this.logger.debug({ currentCapacity });
+        if (currentCapacity.certifChallenge_ChallengeId !== currentCapacity.answer_ChallengeId) {
+          // Copy capacity in error
+          const capacityToModify = { ...currentCapacity };
+          capacityToModify.capacity_CertificationChallengeId = 'REPLACE_ME';
+          // Update also challengeId just to show we gound the right match in logs
+          capacityToModify.certifChallenge_ChallengeId = 'REPLACE_ME';
+
+          // Get the right challenge id for this capacity answer
+          const indexOfCorrectChallenge = allCertificationChallenges.findIndex(
+            (certifChallenge) => certifChallenge.challengeId === currentCapacity.answer_ChallengeId,
+          );
+
+          if (indexOfCorrectChallenge !== -1) {
+            // Update capacity with right answer id
+            capacityToModify.capacity_CertificationChallengeId =
+              allCertificationChallenges.at(indexOfCorrectChallenge).id;
+            // Update also certifChallengeChallengeId just to show we found the right match in logs
+            capacityToModify.certifChallenge_ChallengeId =
+              allCertificationChallenges.at(indexOfCorrectChallenge).challengeId;
+            capacitiesToUpdate.push(capacityToModify);
+          } else {
+            // It is not possible that a capacity has a answer but no challenge
+            throw new Error(`Capacity ${capacityToModify.capacity_AnswerId} do not have a corresponding challenge`);
+          }
+
+          // That answer cannot be used anymore
+          allCertificationChallenges.splice(indexOfCorrectChallenge, 1);
+        } else {
+          // We are on a correct certif challenge, forget about her
+          const correctChallenge = allCertificationChallenges.findIndex(
+            (certifChallenge) => certifChallenge.id === currentCapacity.certifChallenge_Id,
+          );
+          // That certif challenge cannot be used anymore
+          allCertificationChallenges.splice(correctChallenge, 1);
+        }
+      }
+
+      // We have to do the update in reverse because certificationChallengeId is the PRIMARY KEY
+      for (const capacityToUpdate of capacitiesToUpdate.reverse()) {
+        await transaction('certification-challenge-capacities')
+          .where('answerId', '=', capacityToUpdate.capacity_AnswerId)
+          .update({
+            certificationChallengeId: capacityToUpdate.capacity_CertificationChallengeId,
+          });
+      }
+
+      this.logger.debug({ capacitiesToUpdate });
+
+      dryRun ? await transaction.rollback() : await transaction.commit();
+
+      return {
+        numberOfUpdates: capacitiesToUpdate.length,
+      };
+    } catch (error) {
+      await transaction.rollback();
+      throw error;
+    }
+  }
+
+  _getCourseIds({ cursorId, batchSize, transaction }) {
```

Comment:
Le underscore est une ancienne convention, et en ES6 une fonction privee est avec `#`

```suggestion
  #getCourseIds({ cursorId, batchSize, transaction }) {
```